### PR TITLE
WIP Ballistic shield proof of concept

### PR DESCRIPTION
**Pull request by Pariah919**
## About The Pull Request
An attempt by me to make a proof of concept shield using coverage.
Sprite is placeholder.

## Why It's Good For The Game
It's a nice proof of concept for use later.

## Changelog
:cl:
add: Proof of concept ballistic shield.
/:cl:

### CODE REVIEW

**Comment by tgstation-server** [comment]
You currently have a negative Fix/Feature pull request delta of -4. Maintainers may close this PR at will. Fixing issues or improving the codebase will improve this score.

**Comment by Pariah919** [comment]
TODO
Coverage doesn't work for me, not quite sure why or if I messed up.
Integrity shows up twice on the ballistic shield, pretty sure it's related to the examine code being 1 to 100.

File: code/game/objects/items/weapons/shields.dm [hunk 1]
```diff
@@ -84,3 +83,50 @@
...
+		if(60 to 80)
+			to_chat(user, "<span class='notice'>It appears in decent condition, with some damage marks.</span>")
+		if(80 to 100)
+			to_chat(user, "<span class='notice'>It appears in perfect condition.</span>")
+
+/obj/item/weapon/shield/ballistic/attackby(obj/item/I, mob/user)
+	if(istype(I, /obj/item/weapon/baton) && world.time >= cooldown)
+		user.visible_message("<span class='warning'>[user] bashes [src] with [I]!</span>")
+		playsound(user.loc, 'sound/effects/shieldbash.ogg', 25, 1)
+		cooldown = world.time + 2.5 SECONDS
```
Comment (psykzz) [inline code/game/objects/items/weapons/shields.dm:128]: We have new cooldown macros from Roh
`COOLDOWN_CHECK(src, COOLDOWN_SHIELD_BASH)` 
`COOLDOWN_START(src, COOLDOWN_SHIELD_BASH, 2.5 SECONDS)`

You'd need to define `COOLDOWN_SHIELD_BASH`
`#define COOLDOWN_SHIELD_BASH 'cooldown_shield_bash'`

**Comment by Pariah919** [comment]
i'll continue this another time, another day.
I got a bit busier than I thought I would.

**Pull request closed without merging**